R/splitByRanks.R

```diff
@@ -121,28 +115,48 @@ setGeneric("splitByRanks",
     SimpleList(ans)
 }
 
-#' @rdname splitByRanks
+#' @rdname agglomerate-methods
 #' @export
-setMethod("splitByRanks", signature = c(x = "SummarizedExperiment"),
-    function(x, ranks = taxonomyRanks(x), na.rm = TRUE, ...){
+setMethod("agglomerateByRanks", signature = c(x = "SummarizedExperiment"),
+    function(x, ranks = taxonomyRanks(x), na.rm = TRUE, as.list = TRUE, ...){
+        browser()
         args <- .norm_args_for_split_by_ranks(na.rm = na.rm, ...)
-        .split_by_ranks(x, ranks, args)
+        if( !.is_a_bool(as.list) ){
+            stop("'as.list' must be TRUE or FALSE.", call. = FALSE)
+        }
+        if( as.list ){
+            .split_by_ranks(x, ranks, args)
+        }
+        else{
+            altExps(x) <- .split_by_ranks(x, ranks, args)
+            x
+        }
     }
 )
 
-#' @rdname splitByRanks
+#' @rdname agglomerate-methods
 #' @export
-setMethod("splitByRanks", signature = c(x = "SingleCellExperiment"),
-          function(x, ranks = taxonomyRanks(x), na.rm = TRUE, ...){
+setMethod("agglomerateByRanks", signature = c(x = "SingleCellExperiment"),
+          function(x, ranks = taxonomyRanks(x), na.rm = TRUE, as.list = TRUE,
+                    ...){
               args <- .norm_args_for_split_by_ranks(na.rm = na.rm, ...)
               args[["strip_altexp"]] <- TRUE
-              .split_by_ranks(x, ranks, args)
+              if( !.is_a_bool(as.list) ){
+                  stop("'as.list' must be TRUE or FALSE.", call. = FALSE)
+              }
+              if( as.list ){
+                  .split_by_ranks(x, ranks, args)
+              }
+              else{
+                  altExps(x) <- .split_by_ranks(x, ranks, args)
+                  x
+              }
           }
 )
 
-#' @rdname splitByRanks
+#' @rdname agglomerate-methods
 #' @export
-setMethod("splitByRanks", signature = c(x = "TreeSummarizedExperiment"),
+setMethod("agglomerateByRanks", signature = c(x = "TreeSummarizedExperiment"),
           function(x, ranks = taxonomyRanks(x), na.rm = TRUE, ...){
               callNextMethod()
           }
```

Comment:
I think method for TreeSE is not needed because TreeSE is SCE (and that function is called)

TreeSE --> SCE (does have altExp) --> SE